examples/shouldi/shouldi/dependency_check.py

```diff
@@ -0,0 +1,68 @@
+import json
+import os
+import tempfile
+from pathlib import Path
+import asyncio
+from typing import Dict, Any
+
+from dffml.df.base import op
+from dffml.df.types import Definition
+
+package_src_dir = Definition(name="package_src_dir", primitive="str")
+dependency_check_output = Definition(
+    name="dependency_check_output", primitive="Dict[str, Any]"
+)
+tempdir = tempfile.TemporaryDirectory()
+
+
+@op(
+    inputs={"pkg": package_src_dir},
+    outputs={"report": dependency_check_output},
+)
+async def run_dependency_check(pkg: str) -> Dict[str, Any]:
+    """
+    CLI usage: dffml service dev run -log debug shouldi.dependency_check:run_dependency_check -pkg .
+    """
+    if Path(pkg).is_file():
+        proc = await asyncio.create_subprocess_exec(
+            "dependency-check.sh",
+            "-f",
+            "JSON",
+            "--out",
+            '"' + os.path.dirname(tempdir.name) + '"',
```

Comment:
The quotes are only needed on the command line. If you're using something like [execve](https://manpage.me/?q=execve) or it's python equivalent, `create_subprocess_exec`, then having a space in one of the arguments will not affect it, therefore we shouldn't add quotes

```suggestion
            os.path.dirname(tempdir.name),
```